### DISCUSSION TITLE
Filter products by number attribute

### DISCUSSION THREAD

**Discussion by JanikHalder** [swellstores/community#67] [Help] [answered]
Hi there,
is it possible to filter products with the swell.products.list() call by an attribute which is a number?

I want to do something like this:

await swell.products.list({
  page,
  limit,
  sort,
  categories, // optional category slug or ID
  $filters: {
    price: [10, 20],     <-- this is what i want to do
    category: ['featured'],
    size: ['small', 'large']
  }
})

but instead of the price I want to filter some attribute like the hight from one value to another one.

Is this possible? I have tried it as it was used with the price, however this seems not to work.

Best

**Comment by ericingram** [comment, marked as answer]
The frontend API doesn't support this yet. We'll add it to the backlog.

**Reply by JanikHalder** [reply]
Hi @ericingram,

thank you for your response.

Is there a workaround for this at the moment? 

Best

**Comment by logeshswell** [comment]
@JanikHalder I can suggest you a workaround for this till then. Let me know if this helps you.

```node
await swell.products.list({
  page,
  limit,
  sort,
  where: {
    age: {  // teenager
      $gte: 13,
      $lte: 19
    }
  }
})
```

**Reply by JanikHalder** [reply]
@logeshswell 

This seems not to work.

I do it that way:

swell.products.list({
      search: searchText,
      where: {
        ef001438: { // name of the attribute
          $gte: 0,
          $lte: 7000
       }
  })

but my products are not shown.

**Reply by logeshswell** [reply]
@JanikHalder Oh, that's weird. I tried replicating your use-case in my store and after confirming that it's working, I shared it here.
Can you please help me with your store-id? I will confirm it once.

**Reply by logeshswell** [reply]
Did you try removing `search` field because I am wondering if there's any value in searchText that results in zero products.

**Reply by JanikHalder** [reply]
My store ID is rowa-moser.

I have tried it without the search field, but this doesn't change anything still get zero products.

**Reply by logeshswell** [reply]
@JanikHalder Yes, you're right.
I also see the same issue with attributes having number type. But, the same query works with a custom model field having number type, which is not ideal in your case.
So, as Eric mentioned, we'll add it to the backlog and update you when it is available.
